Dashboard/pages/price.py

```diff
@@ -170,77 +41,174 @@ def get_encode_price_reading(
+        text_input_placeholder = f"£{float(product_information['price_threshold']):.2f}"
+    new_threshold = st.text_input('Enter a new Threshold (£):',
+                                    placeholder=text_input_placeholder,
+                                    help='Can be left empty to remove threshold!',
+                                    key=f'new_threshold_value_{product_information['product_id']}')
+    st.write('')
+    if st.button('Update Price Alert',
+                 key=f'new_threshold_submit_{product_information['product_id']}'):
+        valid_threshold = False
+        if new_threshold == '':
+            new_threshold = None
+            valid_threshold = True
+        elif can_parse_as_float(new_threshold):
+            new_threshold = float(new_threshold)
+            if new_threshold > 0:
+                valid_threshold = True
+        if not valid_threshold and new_threshold == product_information['price_threshold']:
+            st.warning('New threshold is the same as the old threshold.')
+        elif valid_threshold:
+            change_threshold_in_db(conn, new_threshold,
+                                   product_information['product_id'], st.session_state['email'])
+            st.rerun()
+        elif not valid_threshold and isinstance (new_threshold, float):
+            st.warning('Threshold must be a positive number!')
+        else:
+            st.warning('Invalid threshold, please try again!')
 
-def display_subscribed_product(config: _Environ, product_information: dict) -> None:
+def unsubscribe_from_product_in_db(conn: connection, product_id: int, email: str) -> None:
+    '''Unsubscribe a user from a product, given product_id and user email.'''
+    user_id = get_user_id(conn, email)
+    with get_cursor(conn) as cur:
+        cur.execute('''DELETE FROM subscriptions
+                        WHERE product_id = %s
+                        AND user_id = %s''', (product_id, user_id))
+        conn.commit()
+
+def unsubscribe_from_product(conn: connection, product_id: int) -> None:
+    '''contains the streamlit display logic for unsubscribing a user from a product,
+    given product id..'''
+    if st.button('Unsubscribe', key=f'unsubscribe_button_{product_id}'):
+        unsubscribe_from_product_in_db(conn, product_id,
+                                       st.session_state['email'])
+        st.rerun()
+
+def product_price_change(price_readings: pd.DataFrame) -> int:
+    '''returns an emoji representing the emoji regarding the price.'''
+    if len(price_readings) == 1:
+        return 0
+    price_readings_sorted_by_time = (
+        price_readings.sort_values(by='reading_at', ascending=False)[:2].reset_index())
+    if (price_readings_sorted_by_time.loc[0]['price']
+        > price_readings_sorted_by_time.loc[1]['price']):
+        return 1
+    if (price_readings_sorted_by_time.loc[0]['price']
+        == price_readings_sorted_by_time.loc[1]['price']):
```

Comment:
```suggestion
    if price_readings_sorted_by_time.loc[0]['price']
        > price_readings_sorted_by_time.loc[1]['price']:
        return 1
    if price_readings_sorted_by_time.loc[0]['price']
        == price_readings_sorted_by_time.loc[1]['price']:
```

Reply:
This doesn't work. Because it spans several lines, so it requires the brackets!